### ISSUE TITLE
show "log in" on Register button for unauthenticated users, when login is required?

### ISSUE THREAD

**Issue by mscholtz** [washingtontrails/vms#398]
For WPs like vol vacations where MBP login is required, the text above the button in the main pane is great.  But I just managed to click it without reading that text, so was surprised when it routed me to MBP login and then to form check.

I would suggest two minor additions here:

1) change the text on the button to something like "Log in, then Join"

2) mention in the text that  I may be routed to the form section before completing my registration.

@WaBirder57 @ebskopin please assign priority here.

**Comment by davisagli** [comment]
Your screenshot above is for a user who's already logged in, not a logged-out user which is the scenario you described.

**Comment by mscholtz** [comment]
@davisagli ?? 
it shows "Sign in to my backpack"..

**Comment by davisagli** [comment]
But it also says "Hi, Ethan" ... were you logged in or not?

**Comment by ebskopin** [comment]
I agree with those recommendations.  It would be nice if the button said "Create Account to Join" and if possible prompted the MBP registration Form (https://mybackpack.wta.org/@@register?came_from=http%3A//staging.wta.org%3A6543/workparty/a2eK0000003Oz24/register) rather than the login landing page as it presently does.

**Comment by davisagli** [comment]
@ebskopin that'll send people in the wrong direction if they already have an account. We link to the login page because it has the option to either log in or create an account.

**Comment by mscholtz** [comment]
@davisagli hmm, pretty sure I was not logged in.  Is it possible that it's doing an inconsistent job of reading the session cookies?  I have been logged in as Ethan many times on this machine, including soon after taking this screenshot.

**Comment by davisagli** [comment]
The session is supposed to be cleared when you log out.

**Comment by mscholtz** [comment]
Not sure what's up, then.  That screen looks different than the screenshot
when I'm really logged in, so it does seem it was in a halfway state
somehow..

Is it possible that it behaves differently if you don't manually log out
but the session is expired?

On Mon, Nov 2, 2015 at 3:14 PM, David Glick <notifications@github.com>
wrote:

> The session is supposed to be cleared when you log out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/398#issuecomment-153188195>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by ebskopin** [comment]
Ahh, that makes sense. Hence the "Log in, then Join". Gotcha. 

I agree that some prep for the forms page would be nice.  When I create a MBP account in order to register I Iand on a "This Page Doesn't Exist.." screen so I'm not sure of the flow for a new reigstrant...

**Comment by davisagli** [comment]
@ebskopin that "this page doesn't exist" is because the Plone registration form redirects to /new-member-welcome, which doesn't exist in the mybackpack.wta.org site but does on the real site: http://www.wta.org/new-member-welcome

**Comment by ebskopin** [comment]
Ok. Can safely assume that upon setting up the account a user would be met with the "Required Forms" as the first step of their Vol Vac registration, correct?

**Comment by davisagli** [comment]
No, it doesn't work that way at this point. We pass a return URL from VMS to Plone, but we've only given the Plone login form an awareness of that return URL, not the registration form.
